File: src/test/java/refactown/cleancode/c08boundary/java/GsonJsonParserTest.java
package refactown.cleancode.c08boundary.java;

import static org.junit.jupiter.api.Assertions.*;

import org.junit.jupiter.api.Test;

import refactown.cleancode.c06objects.PersonJ;

class GsonJsonParserTest {

	@Test
    public void toJson() {
        GsonJsonParser parser = new GsonJsonParser();
        PersonJ billPerson = new PersonJ("Bill", 25, "Rio de Janeiro");
        String billJson = "{\"name\":\"Bill\",\"age\":25,\"city\":\"Rio de Janeiro\"}";
        assertEquals(billJson, parser.toJson(billPerson));
    }

    @Test
    public void fromJson() {
    	GsonJsonParser parser = new GsonJsonParser();
        PersonJ billPerson = new PersonJ("Bill", 25, "Rio de Janeiro");
        String billJson = "{\"name\":\"Bill\",\"age\":25,\"city\":\"Rio de Janeiro\"}";
        assertEquals(billPerson, parser.fromJson(billJson, PersonJ.class));
    }
}

File: src/test/kotlin/refactown/cleancode/c08boundary/GsonJsonParserTest.kt
package refactown.cleancode.c08boundary

import org.junit.jupiter.api.Test

import org.junit.jupiter.api.Assertions.*
import refactown.cleancode.c06objects.Person

internal class GsonJsonParserTest {

    @Test
    fun toJson() {
        val parser = GsonJsonParser()
        val billPerson = Person("Bill", 25, "Rio de Janeiro")
        val billJson = "{\"name\":\"Bill\",\"age\":25,\"city\":\"Rio de Janeiro\"}"
        assertEquals(billJson, parser.toJson(billPerson))
    }

    @Test
    fun fromJson() {
        val parser = GsonJsonParser()
        val billPerson = Person("Bill", 25, "Rio de Janeiro")
        val billJson = "{\"name\":\"Bill\",\"age\":25,\"city\":\"Rio de Janeiro\"}"
        assertEquals(billPerson, parser.fromJson(billJson, Person::class.java))
    }
}